pkg/services/factory.go

```diff
@@ -132,40 +125,44 @@ func (f *defaultFactory) GetGitClients(ctx context.Context, gpClient gitprovider
 		return nil, nil, fmt.Errorf("error getting target name: %w", err)
 	}
 
-	authSvc, err := f.getAuthService(normalizedUrl, gpClient, params.DryRun)
+	authSvc, err := f.getAuthService(configNormalizedUrl, gpClient, params.DryRun)
 	if err != nil {
 		return nil, nil, fmt.Errorf("error creating auth service: %w", err)
 	}
 
-	var appClient, configClient git.Git
-
-	if !params.IsHelmRepository {
-		// We need to do this even if we have an external config to set up the deploy key for the app repo
-		appRepoClient, appRepoErr := authSvc.CreateGitClient(ctx, normalizedUrl, targetName, params.Namespace, params.DryRun)
-		if appRepoErr != nil {
-			return nil, nil, appRepoErr
+	// Do not add deploy key for helm repo, empty url or if its gonna be added below
+	if !params.IsHelmRepository && params.URL != "" && params.URL != params.ConfigRepo {
+		normalizedUrl, err := gitproviders.NewRepoURL(params.URL)
+		if err != nil {
+			return nil, nil, fmt.Errorf("error normalizing url: %w", err)
 		}
 
-		appClient = appRepoClient
-	}
+		provider := authSvc.GetGitProvider()
 
-	if isExternalConfig {
-		normalizedConfigRepo, err := gitproviders.NewRepoURL(params.ConfigRepo, true)
+		repoVisibility, err := provider.GetRepoVisibility(ctx, normalizedUrl)
 		if err != nil {
-			return nil, nil, fmt.Errorf("error normalizing url: %w", err)
+			return nil, nil, fmt.Errorf("error getting repo visibility: %w", err)
 		}
 
-		configRepoClient, configRepoErr := authSvc.CreateGitClient(ctx, normalizedConfigRepo, targetName, params.Namespace, params.DryRun)
-		if configRepoErr != nil {
-			return nil, nil, configRepoErr
+		if *repoVisibility == gitprovider.RepositoryVisibilityPrivate {
```

Comment:
A comment here about why we are doing this would be helpful. I try to leave comments wherever a non-obvious `if` statement appears.

Reply:
@J-Thompson12 Why only for a private repo?